tests/test_js_files.py

```
@@ -0,0 +1,59 @@
+    # Clean comments, string literals, template literals, and regex literals
+    clean_content = re.sub(r'//.*', '', content)
+    clean_content = re.sub(r'/\*[\s\S]*?\*/', '', clean_content)
+    clean_content = re.sub(r'/(?:\\/|[^/\n])+/[gimsuy]*', '""', clean_content)
+    clean_content = re.sub(r'`(?:\\`|[^`])*`', '""', clean_content)
+    clean_content = re.sub(r'(["\'])(?:(?=(\\?))\2[\s\S])*?\1', '""', clean_content)
+
+    for char in clean_content:
+        if char in pairs.values():
+            stack.append(char)
+        elif char in pairs.keys():
+            if stack and stack[-1] == pairs[char]:
```

Ruff (0.16.2)

[warning] Use a single `if` statement instead of nested `if` statements

Combine `if` statements using `and`

(SIM102)